### PR TITLE
Code cleanup: Use Java 17 features

### PR DESCRIPTION
**Pull request by holgerfriedrich**
Use new language features, mainly instanceof pattern matching.
This change covers itests.

Relates to #3508, #3522, and #3585.

### CODE REVIEW

File: itests/org.openhab.core.config.discovery.tests/src/main/java/org/openhab/core/config/discovery/inbox/DynamicThingUpdateOSGiTest.java [hunk 1]
```diff
@@ -121,7 +121,6 @@ public boolean supportsThingType(ThingTypeUID thingTypeUID) {
                 return thingHandler;
             }
         };
-        return thingHandlerFactory;
```
Comment (J-N-K) [inline itests/org.openhab.core.config.discovery.tests/src/main/java/org/openhab/core/config/discovery/inbox/DynamicThingUpdateOSGiTest.java:124]: I would prefer to keep the old version here, it look cleaner.

**Review by J-N-K — CHANGES_REQUESTED** [review]
Thanks for the huge cleanup. I have only one comment.

File: itests/org.openhab.core.config.discovery.tests/src/main/java/org/openhab/core/config/discovery/inbox/DynamicThingUpdateOSGiTest.java (hunk 1 above)
Comment (holgerfriedrich) [reply itests/org.openhab.core.config.discovery.tests/src/main/java/org/openhab/core/config/discovery/inbox/DynamicThingUpdateOSGiTest.java]: ok, rolled back

**Review by J-N-K — APPROVED** [review]
Thanks!

**Pull request merged**